lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -88,28 +94,65 @@ public ReportUserIdentityHandler() {
     public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
             throws IOException {
         LogHelper.attachComponentId(configService);
-
         LOGGER.info(LogHelper.buildLogMessage("Processing report."));
 
+        ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult =
+                ReportProcessingResult.builder();
         try {
-            var userIds =
-                    vcStoreItemDataStore.getItems().stream()
-                            .map(VcStoreItem::getUserId)
-                            .distinct()
-                            .toList();
-            List<ReportUserIdentityItem> reportUserIdentities =
-                    userIds.stream()
-                            .map(
-                                    usrId ->
-                                            new ReportUserIdentityItem(
-                                                    usrId,
-                                                    Vot.P0.name(),
-                                                    Collections.emptyList(),
-                                                    false))
-                            .toList();
-            reportUserIdentityDataStore.createOrUpdate(reportUserIdentities);
+            ReportProcessingRequest reportProcessingRequest =
+                    objectMapper.readValue(inputStream, ReportProcessingRequest.class);
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Request received-" + reportProcessingRequest.toString()));
+            List<String> userIds = new ArrayList<>();
```

Comment:
It looks like you're still attempting to build up a list of all user ids in this list? That's probably tolerable within memory limits, but should be unnecessary - internally we only need to inspect one page of scan results at a time.

Reply:
yeah missed that....updated the logic for processUsersToFindLOCAndUpdateDb() to pick unique users paginated scan from new table.